api/api.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"context"
+
+	"github.com/ONSdigital/dp-search-query/elasticsearch"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/gorilla/mux"
+)
+
+var httpServer *server.Server
+
+//SearchQueryAPI provides an API around elasticseach
+type SearchQueryAPI struct {
+	Router *mux.Router
+}
+
+// CreateAndInitialise initiates a new Search Query API
+func CreateAndInitialise(bindAddr, elasticSearchAPIURL string, errorChan chan error) {
+	router := mux.NewRouter()
+	// Setup libraries and handlers
+	elasticsearch.Setup(elasticSearchAPIURL)
+	errSearch := SetupSearch()
+	if errSearch != nil {
+		log.ErrorC("Failed to setup search templates", errSearch, log.Data{})
+	}
```

Comment:
Minor point and probably personal preference - I like to have a new line / space between blocks of functionality I find that having the line between makes it clear where blocks of logic begin and end. This looks like 1 large block of logic but its actually 3 separate set up calls.

Again this is my personal pref but I think it important to have well laid out readable code.